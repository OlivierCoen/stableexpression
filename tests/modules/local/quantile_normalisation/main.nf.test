nextflow_process {

    name "Test Process QUANTILE_NORMALISE"
    script "modules/local/quantile_normalisation/main.nf"
    process "QUANTILE_NORMALISE"
    tag "quant_norm"
    tag "module"

    test("Should run without failures") {

        when {
            process {
                """
                meta = [dataset: 'test']
                count_file = file( '$projectDir/tests/test_data/quantile_normalise/count.raw.cpm.csv', checkIfExists: true)
                input[0] = [meta, count_file]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
