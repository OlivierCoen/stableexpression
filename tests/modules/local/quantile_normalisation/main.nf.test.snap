{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "dataset": "test"
                        },
                        "count.raw.cpm.quant_norm.parquet:md5,6c49b5fe4e23e64dbc6cad6355432b49"
                    ]
                ],
                "1": [
                    [
                        "QUANTILE_NORMALISE",
                        "python",
                        "3.12.8"
                    ]
                ],
                "2": [
                    [
                        "QUANTILE_NORMALISE",
                        "pandas",
                        "2.2.3"
                    ]
                ],
                "3": [
                    [
                        "QUANTILE_NORMALISE",
                        "scikit-learn",
                        "1.6.1"
                    ]
                ],
                "4": [
                    [
                        "QUANTILE_NORMALISE",
                        "pyarrow",
                        "19.0.0"
                    ]
                ],
                "counts": [
                    [
                        {
                            "dataset": "test"
                        },
                        "count.raw.cpm.quant_norm.parquet:md5,6c49b5fe4e23e64dbc6cad6355432b49"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-08T14:38:27.184977502"
    }
}